lib/connection.js

```diff
@@ -329,6 +368,133 @@ class EsphomeNativeApiConnection extends EventEmitter {
+            ]),
+            'BluetoothDeviceConnectionResponse'
+        );
+    }
+    async listBluetoothGATTServicesService(address) {
+        if (!this.connected) throw new Error(`Not connected`);
+        if (!this.authorized) throw new Error(`Not authorized`);
+        const message = new pb.BluetoothGATTGetServicesRequest([address]);
+
+        const servicesList = [];
+        const onMessage = (message) => {
+            if (message.address === address)
+                servicesList.push(...message.servicesList);
+        };
+        this.on('message.BluetoothGATTGetServicesResponse', onMessage);
+        await this.sendMessageAwaitResponse(
+            message,
+            'BluetoothGATTGetServicesDoneResponse'
+        ).then(
+            () => {
+                this.off('message.BluetoothGATTGetServicesResponse', onMessage);
+            },
+            (e) => {
+                this.off('message.BluetoothGATTGetServicesResponse', onMessage);
+                throw e;
+            }
+        );
+        return { address, servicesList };
+    }
+
+    async readBluetoothGATTCharacteristicService(address, handle) {
+        if (!this.connected) throw new Error(`Not connected`);
+        if (!this.authorized) throw new Error(`Not authorized`);
+        return await this.sendMessageAwaitResponse(
+            new pb.BluetoothGATTReadRequest([address, handle]),
+            'BluetoothGATTReadResponse'
+        );
+    }
+
+    async writeBluetoothGATTCharacteristicService(
+        address,
+        handle,
+        value,
+        response = false
+    ) {
+        if (!this.connected) throw new Error(`Not connected`);
+        if (!this.authorized) throw new Error(`Not authorized`);
+
+        const message = new pb.BluetoothGATTWriteRequest([
+            address,
+            handle,
+            response,
+            value,
+        ]);
+        if (!response) return this.sendMessage(message);
+
+        return await this.sendMessageAwaitResponse(
+            message,
+            'BluetoothGATTWriteResponse'
+        );
+    }
+
+    async notifyBluetoothGATTCharacteristicService(address, handle) {
+        if (!this.connected) throw new Error(`Not connected`);
+        if (!this.authorized) throw new Error(`Not authorized`);
+        return await this.sendMessageAwaitResponse(
+            new pb.BluetoothGATTNotifyRequest([address, handle, true]),
+            'BluetoothGATTNotifyResponse'
+        );
+    }
+
+    async readBluetoothGATTDescriptorService(address, handle) {
+        if (!this.connected) throw new Error(`Not connected`);
+        if (!this.authorized) throw new Error(`Not authorized`);
+        return await this.sendMessageAwaitResponse(
+            new pb.BluetoothGATTReadDescriptorRequest([address, handle]),
+            'BluetoothGATTReadResponse'
+        );
+    }
+
+    async writeBluetoothGATTDescriptorService(
+        address,
+        handle,
+        value,
+        waitForResponse = false
```

Comment:
rename like in "writeBluetoothGATTCharacteristicService"

response ?

Reply:
I was following the naming from AIOESPHOMEAPI.
https://github.com/esphome/aioesphomeapi/blob/d6fd9b6cfd657f1cad8c911d9f766c03a9c7f85c/aioesphomeapi/client.py#L757
vs
https://github.com/esphome/aioesphomeapi/blob/d6fd9b6cfd657f1cad8c911d9f766c03a9c7f85c/aioesphomeapi/client.py#L807

Let me see if there is a reason for the difference. Also looks like I missed readDescriptor so I'll add it

Reply:
Okay - 
For `writeBluetoothGATTCharacteristicService` the `response` parameter describes if the write should be sent using WRITE or WRITE_NO_RESPONSE. It also controls if the connection should wait for a response.

For `writeBluetoothGATTDescriptorService` the `wait_for_response` parameter just controls if the connection should fire-and-forget, or wait for a response.

Reply:
I'll test what happens if you wait for a response when the characteristic is WRITE_NO_RESPONSE